webpack.config.js

```diff
@@ -0,0 +1,41 @@
+const path = require('path');
```

Comment:
In general there is no any differences between dev and prod bundles with such config. Only 'source-map' disabled.
It would be better to have possibility to configure production bundle (separate webpack file of separate function with appropriate plugins).

```
const getProductionPlugins = () => {
  if (!isDevelop) {
    return [
      new UglifyJsPlugin(),
      new MiniCssExtractPlugin({
        filename: '[name].css',
        chunkFilename: '[id].css'
      }),
      new OptimizeCSSAssetsPlugin({})
    ];
  }
  return [];
};

const getDevelopmentPlugins = () => {
  if (isDevelop) {
    return [
      new webpack.HotModuleReplacementPlugin(),
    ];
  }
  return [];
};
```


```
plugins: [
    new CleanWebpackPlugin(),
    new HtmlWebpackPlugin({
       template: "./src/client/index.html"
    }),
    ...getDevelopmentPlugins(),
    ...getProductionPlugins()
  ]
```


Also such kind of segregation between settings can be implemented in scope of two different webpack files
`webpack.dev.config.js`, `webpack.prod.config.js` with some common entry point with appropriate resolver of ENV mode.


P.S.
I would suggest use first example with functions, it will be enough during MP tasks implementation.

Reply:
Added webpack uglify and it works OOTB for webpack 4 for production mode only (without any additional code). Read https://webpack.js.org/guides/production/ and understood that if to use webpack mode a lot of plugins will work out of the box. Couldn't come up  what to add more so with condition still only source map. Also figured about possibility with merge so there will be no duplicate code (in case of 2 different configs).